*** Classes/UnityAppController.origin.mm	2018-02-09 16:49:42.000000000 +0800
--- Classes/UnityAppController.mm	2018-02-11 17:44:27.000000000 +0800
***************
*** 33,38 ****
--- 33,42 ----
  #include "PluginBase/AppDelegateListener.h"
  #include "UTNotificationsTools.h"
  
+ #import <sys/utsname.h>
+ #import "BLGameSdk.h"
+ #import "BLDataSdk.h"
+ 
  // Set this to 1 to force single threaded rendering
  #define UNITY_FORCE_DIRECT_RENDERING 0
  
***************
*** 230,237 ****
--- 234,250 ----
  
  #endif
  
+ - (BOOL)application:(UIApplication *)application openURL:(nonnull NSURL *)url options:(nonnull NSDictionary<NSString *,id> *)options {
+     
+     [[BLGameSdk defaultGameSdk] didGetToken:url];
+     
+     return YES;
+ }
+ 
  - (BOOL)application:(UIApplication*)application openURL:(NSURL*)url sourceApplication:(NSString*)sourceApplication annotation:(id)annotation
  {
+     [[BLGameSdk defaultGameSdk] didGetToken:url];
+ 
      NSMutableArray* keys    = [NSMutableArray arrayWithCapacity: 3];
      NSMutableArray* values  = [NSMutableArray arrayWithCapacity: 3];
  
***************
*** 252,257 ****
--- 265,277 ----
  {
      return YES;
  }
+ - (NSString*)getDeviceVersion
+ {
+     struct utsname systemInfo;
+     uname(&systemInfo);
+     NSString *deviceVersion = [NSString stringWithCString:systemInfo.machine encoding:NSUTF8StringEncoding];
+     return deviceVersion;
+ }
  
  - (BOOL)application:(UIApplication*)application didFinishLaunchingWithOptions:(NSDictionary*)launchOptions
  {
***************
*** 287,293 ****
      [self selectRenderingAPI];
      [UnityRenderingView InitializeForAPI: self.renderingAPI];
  
-     _window         = [[UIWindow alloc] initWithFrame: [UIScreen mainScreen].bounds];
      _unityView      = [self createUnityView];
  
      [DisplayManager Initialize];
--- 307,324 ----
      [self selectRenderingAPI];
      [UnityRenderingView InitializeForAPI: self.renderingAPI];
  
+     // fit for iphonex
+     NSString *deviceVersion = [self getDeviceVersion];//UIDevice.currentDevice.model; 
+     if ([deviceVersion isEqualToString:@"iPhone10,3"] || [deviceVersion isEqualToString:@"iPhone10,6"])
+     {
+         CGRect bounds = CGRectMake(73, 0, 667, 375);
+         _window  = [[UIWindow alloc] initWithFrame:bounds];
+     }
+     else
+     {
+         _window = [[UIWindow alloc] initWithFrame: [UIScreen mainScreen].bounds];
+     }
+ 
      _unityView      = [self createUnityView];
  
      [DisplayManager Initialize];
***************
*** 306,311 ****
--- 337,345 ----
  - (void)applicationDidEnterBackground:(UIApplication*)application
  {
      ::printf("-> applicationDidEnterBackground()\n");
+ 
+     //BLdata sdk
+     [[BLDataSdk sharedDataSdk] applicationDidEnterBackground];
  }
  
  - (void)applicationWillEnterForeground:(UIApplication*)application
***************
*** 323,328 ****
--- 357,365 ----
  - (void)applicationDidBecomeActive:(UIApplication*)application
  {
      ::printf("-> applicationDidBecomeActive()\n");
+     
+     // bldata sdk
+     [[BLDataSdk sharedDataSdk] applicationDidBecomeActive];
  
      [self removeSnapshotView];
  
